### PR TITLE
Docker composition

### PR DESCRIPTION
**Pull request by groldan**

### CODE REVIEW

**Comment by fxprunayre** [comment]
@groldan should we add the routing app to the composition? https://github.com/geonetwork/geonetwork-microservices/blob/main/modules/services/routing/src/main/java/org/fao/geonet/routing/GnRoutingApp.java With that we could expose GN4, on the same entry point as services ? 

Testing it, so far it works well. On GeoNetwork, there is some errors "mapLoadError" probably related to data directory config. I'm checking it.

**Comment by fxprunayre** [comment]
> Testing it, so far it works well. On GeoNetwork, there is some errors "mapLoadError" probably related to data directory config. I'm checking it.

It is because we are using alpha1 for now. Will be fixed when 4.0.0 is released probably next week.

**Comment by groldan** [comment]
> @groldan should we add the routing app to the composition? https://github.com/geonetwork/geonetwork-microservices/blob/main/modules/services/routing/src/main/java/org/fao/geonet/routing/GnRoutingApp.java With that we could expose GN4, on the same entry point as services ?

Done. Renamed as `gateway-service` and got the routes configured declaratively (see `config/gateway-service.yml`)

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    image: gn-cloud-searching:${TAG}
+    depends_on:
+      - config
+      - rabbitmq
+    environment:
+      EUREKA_SERVER_URL: ${EUREKA_SERVER_URL}
+      RABBITMQ_HOST: rabbitmq
+    networks:
+      - gn-cloud-network
+
```
Comment (fxprunayre) [inline docker-compose.yml:184]: Indexing is missing right?

```suggestion
  index:
    image: gn-cloud-indexing:${TAG}
    depends_on:
      - config
      - rabbitmq
    environment:
      EUREKA_SERVER_URL: ${EUREKA_SERVER_URL}
      RABBITMQ_HOST: rabbitmq
    networks:
      - gn-cloud-network
```

**Comment by fxprunayre** [comment]
@groldan I'm getting 

```
gateway_1        | 2020-10-21 06:56:42.750 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application run failed
gateway_1        | 
gateway_1        | java.lang.IllegalStateException: No instances found of configserver (config-service)
``` 

I think it was ok the first time I started (not 100% sure) but now I've it. Any idea of what is wrong?

File: modules/support-services/configuring/src/main/resources/config/geonetwork-dev.properties [hunk 2]
```diff
@@ -1,14 +0,0 @@
-jwt.secret=geonetwork
```
Comment (fxprunayre) [inline modules/support-services/configuring/src/main/resources/config/geonetwork-dev.properties:1]: Where are those files now ? I don't get it.

Comment (groldan) [reply modules/support-services/configuring/src/main/resources/config/geonetwork-dev.properties:1]: `conf/application.yml`. That one contains common config for all services. Additional settings go on each service's `<service-name>.yml`

**Comment by groldan** [comment]
> I think it was ok the first time I started (not 100% sure) but now I've it. Any idea of what is wrong?

With all that's going during a full startup, the config-service just wasn't ready when a dependant service tried to contact it and timed out. You can just run `docker-compose up -d` again to relaunch all the ones that exited. 
Usually I use [dockerize](https://github.com/jwilder/dockerize) in the service `command` in docker-compose.yml to wait for required services to be ready before launching a given one. See for example https://github.com/camptocamp/geoserver-microservices/blob/master/docker-compose.yml#L121
Only trouble is since we're using the new buildpacks support in spring boot 2.3 to build the docker images instead of a regular Dockerfile, I don't yet know how to install dockerize in the target images.

**Comment by fxprunayre** [comment]
> You can just run `docker-compose up -d` again to relaunch all the ones that exited.

Yes indeed. It works.

Also for starting another service as local, I've been using
```shell
SERVER_PORT=9901 mvn spring-boot:run -Dspring-boot.run.profiles=dev,local -f modules/services/searching
```
because it was trying to startup on 8080 which is used by geonetwork container.

**Pull request merged**